### DISCUSSION TITLE
Six pack of "waterproof" DS18B20 with 3m cables bought end of 2023

### DISCUSSION THREAD

**Discussion by rmk92** [cpetrich/counterfeit_DS18B20#36] [General]
I bought a pack of waterproof DS18B20 with 3m cables off a large online retailer at the end of 2023.

The first clue that something strange was going on was that one of these had an entirely zero ROM code, meaning it fails the CRC check.

Upon finding your project here, I threw code into my ESP32 program to read the various values, and I get:
68=ff 93=ff 97=ff 8b=00
which seems to suggest a D2 family sensor. This is further confirmed when reading the scratchpad registers:
90 01 55 05  7f a5 a5 66  b4

The five that "work" have family/serial numbers reported by Linux as 28-3ce10457xxxx.

I also have a set of three 1m versions bought previously that have family/serial numbers of the form 28-3cxxe381xxxx. I haven't set about attempting to identify these yet.

**Comment by cpetrich** [comment]
thank you for testing and sharing more recent ROM codes of Family D2.

**Comment by bakup** [comment]
2024
28-EF-B0-45-D4-07-23-15:
Scratchpad Register: 90/01/00/00/7F/00/00/00/C9
28-15-62-57-04-E1-3C-50: 
Scratchpad Register: 90/01/00/00/7F/00/00/00/C9
